python/podio_schema_evolution.py

```diff
@@ -327,6 +425,33 @@ def heuristics(self):
         added_members = [change for change in schema_changes if isinstance(change, AddedMember)]
         self.heuristics_members(added_members, dropped_members, schema_changes)
 
+        for vmc in (c for c in schema_changes if isinstance(c, AddedVectorMember)):
+            self.errors.append(
+                f"Forbidden schema change in '{vmc.klassname}': Added vector member '{vmc.member}'"
+            )
+        for vmc in (c for c in schema_changes if isinstance(c, DroppedVectorMember)):
+            self.errors.append(
+                f"Forbidden schema change in '{vmc.klassname}': Added vector member '{vmc.member}'"
+            )
+
+        for rc in (c for c in schema_changes if isinstance(c, AddedSingleRelation)):
+            self.errors.append(
+                f"Forbidden schema chage in '{rc.klassname}': Added OneToOneRelation '{rc.member}'"
+            )
+        for rc in (c for c in schema_changes if isinstance(c, DroppedSingleRelation)):
+            self.errors.append(
+                f"Forbidden schema chage in '{rc.klassname}': Dropped OneToOneRelation '{rc.member}'"
+            )
+
+        for rc in (c for c in schema_changes if isinstance(c, AddedMultiRelation)):
+            self.errors.append(
+                f"Forbidden schema chage in '{rc.klassname}': Added OneToManyRelation '{rc.member}'"
+            )
+        for rc in (c for c in schema_changes if isinstance(c, DroppedMultiRelation)):
+            self.errors.append(
+                f"Forbidden schema chage in '{rc.klassname}': Dropped OneToManyRelation '{rc.member}'"
+            )
+
```

Comment:
Might be worth moving the error message into the specific schema change type for further simplification.
```suggestion
        for c in schema_changes:
            if isinstance(c, AddedVectorMember):
                self.errors.append(
                    f"Forbidden schema change in '{c.klassname}': Added vector member '{c.member}'"
                )
            if isinstance(c, DroppedVectorMember):
                self.errors.append(
                    f"Forbidden schema change in '{c.klassname}': Dropped vector member '{c.member}'"
               )
            if isinstance(c, AddedSingleRelation):
                self.errors.append(
                    f"Forbidden schema change in '{c.klassname}': Added OneToOneRelation '{c.member}'"
                )
            if isinstance(c, DroppedSingleRelation):
                self.errors.append(
                    f"Forbidden schema change in '{c.klassname}': Dropped OneToOneRelation '{cc.member}'"
               )
            if isinstance(c, AddedMultiRelation):
                self.errors.append(
                    f"Forbidden schema change in '{c.klassname}': Added OneToManyRelation '{c.member}'"
                )
            if isinstance(c, DroppedMultiRelation):
                self.errors.append(
                    f"Forbidden schema change in '{c.klassname}': Dropped OneToManyRelation '{c.member}'"
                )
```

Reply:
Actually we already have a message in the schema changes, so simply reusing that is indeed the way to go here. Thanks for the hint (trees and woods and such from my side).